builder/fetcher.go

```diff
@@ -0,0 +1,58 @@
+package builder
+
+import (
+	"bytes"
+	"log"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/ercole-io/ercole-agent/config"
+)
+
+func fetcher(configuration config.Configuration, fetcherName string, params ...string) []byte {
+	var (
+		cmd    *exec.Cmd
+		err    error
+		psexe  string
+		stdout bytes.Buffer
+		stderr bytes.Buffer
+	)
+
+	baseDir := config.GetBaseDir()
+
+	if runtime.GOOS == "windows" {
+		psexe, err = exec.LookPath("powershell.exe")
+		if err != nil {
+			log.Fatal(psexe)
+		}
+		if configuration.ForcePwshVersion == "0" {
+			params = append([]string{"-ExecutionPolicy", "Bypass", "-File", baseDir + "\\fetch\\win.ps1", "-s", fetcherName}, params...)
+		} else {
+			params = append([]string{"-version", configuration.ForcePwshVersion, "-ExecutionPolicy", "Bypass", "-File", baseDir + "\\fetch\\win.ps1", "-s", fetcherName}, params...)
+		}
+		log.Println("Fetching " + psexe + " " + strings.Join(params, " "))
+
+		cmd = exec.Command(psexe, params...)
+	} else {
+		log.Println("Fetching " + baseDir + "/fetch/" + fetcherName + " " + strings.Join(params, " "))
+		cmd = exec.Command(baseDir+"/fetch/"+fetcherName, params...)
+	}
+
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err = cmd.Run()
+	if len(stderr.Bytes()) > 0 {
+		log.Print(string(stderr.Bytes()))
+	}
+
+	if err != nil {
+		if fetcherName != "dbstatus" {
+			log.Fatal(err)
```

Comment:
What happen when a gorouting panic()?

Reply:
If a goroutine panic, the execution will stall.
This is surely a side effect we should avoid managing panic.
I'll have to manage this with errgroups https://godoc.org/golang.org/x/sync/errgroup

Reply:
Or maybe we should avoid to panic and manage errors: it's easier and prettier.

Reply:
Mhmhmh

Reply:
`os.Exit` cause the entire process to exit, so `log.Fatal` is not a problem.

Reply:
https://play.golang.org/p/rwofMzlOLKY
Both panic and log.Fatal terminate the program.
If we use panic we can recover and manage the error inside the goroutine, but if before wasn't managed I don't think I should did it here.